lldb/tools/lldb-server/lldb-platform.cpp

```diff
@@ -301,13 +294,31 @@ int main_platform(int argc, char *argv[]) {
       exit(socket_error);
     }
     printf("Connection established.\n");
+
+    std::optional<uint16_t> port = 0;
```

Comment:
If we're going to use optional then this should be left to default construct to std::nullopt (https://en.cppreference.com/w/cpp/utility/optional/nullopt_t) (the equivalent of Haskell Maybe's "Nothing" state).

Reply:
I removed the optional port here, and just set the portmap within the availablePort branch.

I also realized I forgot to take into account the non-server mode, so I put `platform.SetPortMap(std::move(gdbserver_portmap));` for the else branch where `lldb-server` isn't running with the `--server` flag.